app/controllers/store_controller.rb

```diff
@@ -217,7 +217,11 @@ def place_order
     end
     unless @order.ready_for_purchase?
       flash[:alert] = @order.errors.as_html
-      redirect_to_checkout
+      unless @recipient.valid_as_purchaser?
```

Comment:
Why is there a change in the process of customer's purchase process?

Reply:
I didn't actually change anything, I think I was adding and deleting stuff trying to fix a particularly crazy bug and then I completely refactored to avoid the problem later. Store-controller is not changed overall.